web/src/navbar_menus.js

```diff
@@ -4,54 +4,101 @@ import {page_params} from "./page_params";
 import * as personal_menu_popover from "./personal_menu_popover";
 import * as popover_menus from "./popover_menus";
 
+// Available menu options based on user type
+const popover_menus_options = page_params.is_spectator
+    ? ["signup_button", "login_button", "navbar_help_menu", "gear_menu"]
+    : ["navbar_help_menu", "gear_menu", "personal_menu"];
+
 export function is_navbar_menus_displayed() {
     return (
-        popover_menus.is_personal_menu_popover_displayed() ||
-        popover_menus.is_gear_menu_popover_displayed() ||
-        popover_menus.is_help_menu_popover_displayed()
+        document.activeElement &&
+        (document.activeElement.classList.contains("signup_button") ||
+            document.activeElement.classList.contains("login_button") ||
+            popover_menus.is_personal_menu_popover_displayed() ||
+            popover_menus.is_gear_menu_popover_displayed() ||
+            popover_menus.is_help_menu_popover_displayed())
     );
 }
 
+// Handle keyboard events for navigation
 export function handle_keyboard_events(event_name) {
-    // We don't need to process arrow keys in navbar menus for spectators
-    // since they only have gear menu present.
-    if (
-        popover_menus.is_personal_menu_popover_displayed() &&
-        (event_name === "left_arrow" || event_name === "gear_menu") &&
-        !page_params.is_spectator
-    ) {
-        // Open gear menu popover on left arrow.
-        personal_menu_popover.toggle();
-        gear_menu.toggle();
-        return true;
+    let i = 0;
+    for (const selected_popover of popover_menus_options) {
+        if (is_menu_displayed(selected_popover)) {
+            return handle_menu_event(selected_popover, event_name, i);
+        }
+        i = i + 1;
```

Comment:
This can be writtten as:
```js
export function handle_keyboard_events(event_name) {
    const index = popover_menus_options.findIndex(is_menu_displayed);

    if (index !== -1) {
        return handle_menu_event(popover_menus_options[index], event_name, index);
    }

    return false;
}
```

Reply:
Updated.